File: backend/src/main/java/com/hospital/backend/entity/Appointment.java
package com.hospital.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "appointments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", nullable = false)
    private User patient;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctor_id", nullable = false)
    private Doctor doctor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "availability_id", nullable = false)
    private DoctorAvailability doctorAvailability;

    @Column(nullable = false)
    private LocalDateTime appointmentDate;

    // PENDING, CONFIRMED, REJECTED, COMPLETED, CANCELLED
    @Column(nullable = false, length = 20)
    private String status;

    private String reason;
}

File: frontend/src/services/appointment.js
import api from './auth';

class AppointmentService {
    getAllAppointments() {
        return api.get('/appointments');
    }

    getAppointmentsByPatient(patientId) {
        return api.get(`/appointments/patient/${patientId}`);
    }

    getAppointmentsByDoctor(doctorId) {
        return api.get(`/appointments/doctor/${doctorId}`);
    }

    bookAppointment(patientId, doctorId, availabilityId, reason) {
        return api.post(`/appointments/book?patientId=${patientId}&doctorId=${doctorId}&availabilityId=${availabilityId}`, { reason });
    }

    updateStatus(appointmentId, status) {
        return api.put(`/appointments/${appointmentId}/status`, { status });
    }
}

export const appointmentService = new AppointmentService();
